11 service activity

sschoetz opened the pull request:
Implement activity service.
Closes #11

sschoetz committed:
Implement update and insert
functions for  activity service.

sschoetz committed:
Merge remote-tracking branch 'origin/develop' into 11-service-activity

sschoetz committed:
Partial implementation of activity-service.

sschoetz committed:
Add testing dependencies.

sschoetz committed:
Merge remote-tracking branch 'origin/develop' into 11-service-activity

sschoetz committed:
Add additional tests.

sschoetz committed:
Implement getBetween and getByCategory
for aws-avtivity service.

sschoetz committed:
Merge remote-tracking branch 'origin/develop' into 11-service-activity

sschoetz committed:
implement activity-service subscribe.

sschoetz committed:
Merge remote-tracking branch 'origin/develop' into 11-service-activity

sschoetz committed:
Ignore laucher for debug.

sschoetz committed:
Implement notification on activity change.

sschoetz committed:
Run prettier.

sschoetz committed:
Merge remote-tracking branch 'origin/develop' into 11-service-activity

AndHager commented on ctt-frontend/.eslintignore:
```diff
@@ -1,3 +1,4 @@
-coverage/
\ No newline at end of file
+coverage/
+src\models
```
```suggestion
src/models
```

sschoetz committed:
Extract Subscriber class and observables to
separate files and fix subscribing with username.
Also disable subscribing for tests.

sschoetz committed:
Merge remote-tracking branch 'origin/develop' into 11-service-activity

sschoetz committed:
Update ctt-frontend/.eslintignore

Co-authored-by: AndHager <ahager@hm.edu>

walli545 commented on ctt-frontend/src/app/services/aws-activity.service.ts:
```diff
@@ -1,73 +1,255 @@
+      },
+      () =>
+        Promise.reject(
+          `Could nor insert Activity at starttime ${insert.from.toISOString()}`
```
```suggestion
          `Could not insert Activity at starttime ${insert.from.toISOString()}`
```

walli545 committed:
Update ctt-frontend/src/app/services/aws-activity.service.ts

walli545 committed:
Fix prettier and remove comments

walli545 reviewed (changes requested):
I fixed the prettier error, but there are still warnings when serving the app. Can you please take a look at them?
```
Error: src/app/app.module.ts:133:43 - error NG2005: The class 'AwsActivityService' cannot be created via dependency injection, as it does not have an Angular decorator. This will result in an error at runtime.

Either add the @Injectable() decorator to 'AwsActivityService', or configure a different provider (such as a provider with 'useFactory').


133     { provide: ActivityService, useClass: AwsActivityService },
                                              ~~~~~~~~~~~~~~~~~~

  src/app/services/aws-activity.service.ts:21:1
     21 @Injectable({
        ~~~~~~~~~~~~~
     22   providedIn: 'root',
        ~~~~~~~~~~~~~~~~~~~~~
    ... 
    248   }
        ~~~
    249 }
        ~
    'AwsActivityService' is declared here.


Error: src/app/services/aws-activity.service.ts:28:40 - error NG2003: No suitable injection token for parameter 'subscribe' of class 'AwsActivityService'.
  Consider using the @Inject decorator to specify an injection token.

28   constructor(private api: APIService, subscribe: boolean = true) {
                                          ~~~~~~~~~

1m  src/app/services/aws-activity.service.ts:28:51
    28   constructor(private api: APIService, subscribe: boolean = true) {
                                                         ~~~~~~~
    This type is not supported as injection token.
```

sschoetz committed:
Inject subscribe parameter for activity service.

sschoetz committed:
Merge branch '11-service-activity' of https://github.com/electron2302/continuous-time-tracking into 11-service-activity

sschoetz committed:
Inject boolean for activity service.

sschoetz committed:
Run prettier.

sschoetz committed:
fix typo in testing.

sschoetz committed:
Merge remote-tracking branch 'origin/develop' into 11-service-activity
